pkg/operator/target_config_reconciler.go

```diff
@@ -55,6 +62,21 @@ func NewTargetConfigReconciler(
 	kubeClient kubernetes.Interface,
 	eventRecorder events.Recorder,
 ) *TargetConfigReconciler {
+	// make sure our list of excluded system namespaces is up to date
+	allNamespaces, err := kubeClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.ErrorS(err, "error listing namespaces")
+		return nil
+	}
+	for _, ns := range allNamespaces.Items {
+		if matched, err := regexp.Match(`^openshift-[\w\d-]*`, []byte(ns.Name)); matched {
```

Comment:
`strings.HasPrefix` might be a bit faster

Reply:
and easier to understand, done